core/indexer/elasticProcessor.go

```diff
@@ -0,0 +1,399 @@
+package indexer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/indexer/workItems"
+	"github.com/ElrondNetwork/elrond-go/core/statistics"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+type elasticProcessor struct {
+	*txDatabaseProcessor
+
+	elasticClient DatabaseClientHandler
+	parser        *dataParser
+}
+
+// NewElasticProcessor creates an elasticsearch es and handles saving
+func NewElasticProcessor(arguments ElasticProcessorArgs) (ElasticProcessor, error) {
+	ei := &elasticProcessor{
```

Comment:
Nil checks?

Reply:
checks are done in factory constructor.

Reply:
Since this constructor is exported, someone else might use is elsewhere and use it improperly.

Reply:
done.